lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -13,548 +13,1055 @@
+        return VerbatimType({SmallString<0>((iface.getValue().getName() + " " +
+                                             bundle->getName().getValue())
+                                                .str()),
+                             true});
+      })
+      .Case<Unsupported<AugmentedType::String>>([&](auto *field) -> TypeSum {
+        return unsupported(field->getName().getValue(), "string");
+      })
+      .Case<Unsupported<AugmentedType::Boolean>>([&](auto *field) -> TypeSum {
+        return unsupported(field->getName().getValue(), "boolean");
+      })
+      .Case<Unsupported<AugmentedType::Integer>>([&](auto *field) -> TypeSum {
+        return unsupported(field->getName().getValue(), "integer");
+      })
+      .Case<Unsupported<AugmentedType::Double>>([&](auto *field) -> TypeSum {
+        return unsupported(field->getName().getValue(), "double");
+      })
+      .Case<Unsupported<AugmentedType::Literal>>([&](auto *field) -> TypeSum {
+        return unsupported(field->getName().getValue(), "literal");
+      })
+      .Case<Unsupported<AugmentedType::Deleted>>([&](auto *field) -> TypeSum {
+        return unsupported(field->getName().getValue(), "deleted");
+      });
 }
 
-/// Remove Grand Central Annotations from ports of modules or external modules.
-/// Return argument attributes with annotations removed.
-void GrandCentralVisitor::handlePorts(Operation *op) {
+/// Traverse an Annotation that is an AugmentedBundleType.  During
+/// traversal, construct any discovered SystemVerilog interfaces.  If this
+/// is the root interface, instantiate that interface in the parent. Recurse
+/// into fields of the AugmentedBundleType to construct nested interfaces
+/// and generate stringy-typed SystemVerilog hierarchical references to
+/// drive the interface. Returns false on any failure and true on success.
+Optional<sv::InterfaceOp>
+GrandCentralPass::traverseBundle(Bundle bundle, IntegerAttr id, Twine path) {
+  auto builder = OpBuilder::atBlockEnd(getOperation().getBody());
+  sv::InterfaceOp iface;
+  builder.setInsertionPointToEnd(getOperation().getBody());
+  auto loc = getOperation().getLoc();
+  auto iFaceName = circuitNamespaceRef->newName(bundle.getName().getValue());
+  iface = builder.create<sv::InterfaceOp>(loc, iFaceName);
+  iface->setAttr(
+      "output_file",
+      hw::OutputFileAttr::get(builder.getStringAttr(getOutputDirectory()),
+                              builder.getStringAttr(iFaceName + ".sv"),
+                              builder.getBoolAttr(true),
+                              builder.getBoolAttr(true), builder.getContext()));
+
+  builder.setInsertionPointToEnd(cast<sv::InterfaceOp>(iface).getBodyBlock());
+
+  for (auto element : bundle.getElements()) {
+    auto field = fromAttr(element);
+    if (!field)
+      return None;
+
+    auto name = element.cast<DictionaryAttr>().getAs<StringAttr>("name");
+    if (!name)
+      name = element.cast<DictionaryAttr>().getAs<StringAttr>("defName");
```

Comment:
Is this an artifact would fix in Chisel or is there a semantic reason why name and defName are in the annotation and we seem to always ignore it?

Reply:
It's an issue with `AugmentedBundleType` using `defName` and every other `Augmented*Type` using `name`. I think I already fixed this with the `getName()` method in `AugmentedType` which is overriden in `AugmentedBundleType`. I can switch to that, just forgot that existed.